samples/app-anonymous-users/csharp/Controllers/AuthController.cs

```diff
@@ -0,0 +1,115 @@
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json;
+using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace AnonymousUsers.Controllers
+{
+    public class AuthController : Controller
+    {
+        /// <summary>
+        /// Represents a set of key/value application configuration properties.
+        /// </summary>
+        public readonly IConfiguration _configuration;
+        private readonly IHttpClientFactory _httpClientFactory;
+        private readonly IHttpContextAccessor _httpContextAccessor;
+
+        public AuthController(
+            IConfiguration configuration,
+            IHttpClientFactory httpClientFactory,
+            IHttpContextAccessor httpContextAccessor)
+        {
+            _configuration = configuration;
+            _httpClientFactory = httpClientFactory;
+            _httpContextAccessor = httpContextAccessor;
+        }
+
+        /// <summary>
+        /// Get facebook profile of user.
+        /// </summary>
+        /// <param name="accessToken">Token</param>
+        /// <returns></returns>
+        [HttpPost]
+        [Route("getFbAccessToken")]
+        public async Task<JsonResult> GetFacebookAuthToken(string accessToken)
+        {
+            var fbAppId = _configuration["FacebookAppId"];
+            var fbPassword = _configuration["FacebookAppPassword"];
+            var redirectUrl = _configuration["ApplicationBaseUrl"] + "/facebook-auth-end";
+
+            var client = new HttpClient();
+            string responseBody;
+            var response = await client.GetAsync(string.Format("https://graph.facebook.com/v12.0/oauth/access_token?client_id={0}&redirect_uri={1}&client_secret={2}&code={3}", fbAppId, redirectUrl, fbPassword, accessToken));
+
+            if (response.IsSuccessStatusCode)
+            {
+                responseBody = await response.Content.ReadAsStringAsync();
+                var token = JsonConvert.DeserializeObject<dynamic>(responseBody).access_token;
+
+                var profile = FacebookHelper.GetUri("https://graph.facebook.com/me",
+                Tuple.Create("fields", "name,picture"),
+                Tuple.Create("access_token", token.Value));
+
+                var res = await FacebookHelper.FacebookRequest<FacebookProfile>(profile);
+                var result = new
+                {
+                    name = res.Name,
+                    picture = res.ProfilePicture.data.url
+                };
+
+                var jsonString = JsonConvert.SerializeObject(result);
+
+                return Json(jsonString);
+            }
+            else
+            {
+                return Json("Error occoured");
+            }
+        }
+
+        /// <summary>
+        /// Get user access token
+        /// </summary>
+        /// <returns></returns>
+        [HttpGet]
+        [Route("GetUserAccessToken")]
+        public async Task<ActionResult<UserData>> GetUserAccessToken()
```

Comment:
Similarly this doesn't return an access token, please rename to match it's actual purpose

Reply:
Updated your feedback.